### ISSUE TITLE
Proper Accessibility for link buttons

### ISSUE THREAD

**Issue by doug-s-nava** [HHS/simpler-grants-gov#9169]
### Summary

Nesting a button inside of a link creates a situation where assistive technologies may not know which actionable element to use. Instances of this pattern in the app should be updated to use a `<Link className="usa-button">` pattern instead

### Acceptance criteria

- [ ] all instances of buttons nested inside `<Link>` or `<a>` components are updated to use a non-nested `<Link className="usa-button">` pattern instead
- [ ] all link buttons in the app focus correctly when using keyboard based navigation
- [ ] screen reader interprets all link buttons in the app correctly
